### PR TITLE
Run Content infra tests with each SDK build

### PR DESCRIPTION
**Pull request by maimorag**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-5568

## Description
We have content unit tests that rely on SDK changes.
This PR add a step to the SDK build (running on every PR), at the root (top) level, that verifying unit tests wouldn't fail after changes in sdk

### CODE REVIEW

File: .github/workflows/validate-changelog-content-infra-tests.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        uses: actions/checkout@v3
+
+      - name: Setup Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: '3.10'
+
+      - name:  Get commit hash
+        id: get-commit
+        run: echo commit=$(git rev-parse HEAD) >> $GITHUB_OUTPUT
```
Comment (dorschw) [inline .github/workflows/validate-changelog-content-infra-tests.yml:19]: is this needed, or just for debugging?

File: .github/workflows/validate-changelog-content-infra-tests.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+
+      - name: Setup Poetry
+        uses: Gr1N/setup-poetry@v8
+
+      - name: Downloding the last demisto-sdk version
+        run: |
+          cd content
+          poetry install --with ci
+          echo "starting uninstall of demisto-sdk .. "
```
Comment (dorschw) [inline .github/workflows/validate-changelog-content-infra-tests.yml:35]:
```suggestion
          echo "uninstalling content-master version of demisto-sdk"
```

File: .github/workflows/validate-changelog-content-infra-tests.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+          pip uninstall -y demisto-sdk
+          echo "uninstall of demisto-sdk is done .. "
```
Comment (dorschw) [inline .github/workflows/validate-changelog-content-infra-tests.yml:37]:
```suggestion
          echo "uninstalled content-master demisto-sdk"
```

File: .github/workflows/validate-changelog-content-infra-tests.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+          echo "starting installing custom demisto-sdk of commit: ${{ steps.get-commit.outputs.commit }} "
```
Comment (dorschw) [inline .github/workflows/validate-changelog-content-infra-tests.yml:38]:
```suggestion
          echo "installing branch version of demisto-sdk (${{ steps.get-commit.outputs.commit }} )"
```

File: .github/workflows/validate-changelog-content-infra-tests.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+          pip install git+https://github.com/demisto/demisto-sdk@${{ steps.get-commit.outputs.commit }}
+          echo "added custom SDK"
```
Comment (dorschw) [inline .github/workflows/validate-changelog-content-infra-tests.yml:40]:
```suggestion
```

File: .github/workflows/validate-changelog-content-infra-tests.yml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+      - name: tests
```
Comment (dorschw) [inline .github/workflows/validate-changelog-content-infra-tests.yml:42]:
```suggestion
      - name: Content Unit Tests 
```

File: .github/workflows/validate-changelog-content-infra-tests.yml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        env:
+          PYTHONPATH: ${{ github.workspace }}/content
+        run: |
+          cd content
+          poetry run pytest ./Tests/scripts/infrastructure_tests/ -v
+          poetry run pytest ./Tests/Marketplace/Tests/ -v
+          poetry run pytest ./Tests/tests -v
+          poetry run pytest ./Tests/private_build/tests -v
+          poetry run pytest Utils -v
```
Comment (dorschw) [inline .github/workflows/validate-changelog-content-infra-tests.yml:51]: can we run them all in one command?

File: .github/workflows/validate-changelog-content-infra-tests.yml [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        uses: actions/checkout@v3
+        with:
+          repository: demisto/content
+          path: content
+
+
+      - name: Setup Poetry
+        uses: Gr1N/setup-poetry@v8
+
+      - name: Downloding the last demisto-sdk version
```
Comment (dorschw) [inline .github/workflows/validate-changelog-content-infra-tests.yml:31]:
```suggestion
      - name: Install demisto-SDK from branch
```

File: .github/workflows/validate-changelog-content-infra-tests.yml (hunk 1 above)
Comment (maimorag) [reply .github/workflows/validate-changelog-content-infra-tests.yml:19]: needed

File: .github/workflows/validate-changelog-content-infra-tests.yml (hunk 7 above)
Comment (maimorag) [reply .github/workflows/validate-changelog-content-infra-tests.yml:50]: no, it didn't work

File: .github/workflows/validate-changelog-content-infra-tests.yml [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+      - name: Setup Poetry
+        uses: Gr1N/setup-poetry@v8
+
+      - name: Install demisto-SDK from branch
+        run: |
+          cd content
+          poetry install --with ci
+          echo "uninstalling content-master version of demisto-sdk"
+          pip uninstall -y demisto-sdk
+          echo "uninstalled content-master demisto-sdk"
```
Comment (Shellyber) [inline .github/workflows/validate-changelog-content-infra-tests.yml:37]: These 2 echos are the same correct me if I'm wrong.
If so, let's remove the second one (after the command you run)
And change the first one to: `Uninstalling demisto-sdk from content master`, I think it is clearer

File: .github/workflows/validate-changelog-content-infra-tests.yml [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+          echo "installing branch version of demisto-sdk (${{ steps.get-commit.outputs.commit }} )"
+          pip install git+https://github.com/demisto/demisto-sdk@${{ steps.get-commit.outputs.commit }}
+
+      - name: Content Unit Tests
+        env:
+          PYTHONPATH: ${{ github.workspace }}/content
+        run: |
+          cd content
```
Comment (Shellyber) [inline .github/workflows/validate-changelog-content-infra-tests.yml:45]:
```suggestion
          cd content
          echo "Running all unit tests"
```

**Comment by Shellyber** [comment]
@maimorag @dorschw How much time it adds to the SDK PRs?
Also, @maimorag Please provide a PR example where you ran this flow for testing

**Comment by dorschw** [comment]
> @maimorag @dorschw How much time it adds to the SDK PRs? Also, @maimorag Please provide a PR example where you ran this flow for testing

image
Seems like it takes ~5 minutes, and is in parallel to the usual (~15 minute) SDK build, so it doesn't add any really, the build will still finish in ~15 minutes.

File: .github/workflows/validate-changelog-content-infra-tests.yml [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+      - name: Setup Poetry
+        uses: Gr1N/setup-poetry@v8
+
+      - name: Install demisto-SDK from branch
+        run: |
+          cd content
+          poetry install --with ci
+          echo "uninstalling content-master version of demisto-sdk"
+          pip uninstall -y demisto-sdk
```
Comment (dorschw) [inline .github/workflows/validate-changelog-content-infra-tests.yml:36]: See [this](https://github.com/demisto/demisto-sdk/actions/runs/5641660966/job/15280070757?pr=3369#step:7:214) warning, I'm not sure this is running in the poetry venv. Let's try `poetry run pip uninstall demisto-sdk`, or `poetry remove demisto-sdk` (could be slower, but is more likely to succeed)

Comment (maimorag) [reply .github/workflows/validate-changelog-content-infra-tests.yml]: I used poetry remove demisto-sdk

**Comment by maimorag** [comment]
commit 0c139c923a22dbcb77e99666f2eb6096f31970cc is an example where you can catch an error that would fail this step since it causes unit test to fail

**Review by dorschw — APPROVED** [review]

**Pull request merged**